lib/ts/utils.ts

```diff
@@ -80,7 +80,18 @@ export function getRedirectToPathFromURL(): string | undefined {
         try {
             const normalisedURLPath = new NormalisedURLPath(param).getAsStringDangerous();
             const pathQueryParams = param.split("?")[1] !== undefined ? `?${param.split("?")[1]}` : "";
-            return normalisedURLPath + pathQueryParams;
+            const pathWithQueryParams = normalisedURLPath + pathQueryParams;
+
+            // Ensure a leading "/" if `normalisedUrlPath` is empty but `pathWithQueryParams` is not to ensure proper redirection.
+            // Example: "?test=1" will not redirect the user to `/?test=1` if we don't add a leading "/".
+            if (
+                normalisedURLPath.length === 0 &&
+                pathWithQueryParams.length > 0 &&
+                !pathWithQueryParams.startsWith("/")
+            ) {
+                return "/" + pathWithQueryParams;
+            }
+            return pathWithQueryParams;
```

Comment:
this is a bug fix from before right?

Reply:
If so, and previously, if there were no query params, and the redirectToPath was `/`, we would end up redirecting to `""`? How did that work?

Reply:
If the `normalisedURLPath` was empty then it would be considered as if there was no `redirectToPath` and we would end up doing the default `/` redirection. 

The bug presents itself when we preserve query params because now `redirecToPath` exists but without a leading slash.